onnxruntime/core/framework/sequential_executor.cc

```diff
@@ -122,11 +123,187 @@ static Status ReleaseNodeMLValues(ExecutionFrame& frame,
+                                          const std::vector<OrtValue>& feeds, const std::vector<int>& fetch_mlvalue_idxs,
+                                          std::vector<OrtValue>& fetches, const std::unordered_map<size_t, IExecutor::CustomAllocator>& fetch_allocators,
+                                          const logging::Logger& logger, int64_t run_id) {
+  const SequentialExecutionPlan& seq_exec_plan = *session_state.GetExecutionPlan();
+  const auto& exec_plan_vec = seq_exec_plan.execution_plan;
+  const auto exec_plan_last_index = exec_plan_vec.size() - 1;
+  std::shared_ptr<ExecutionFrame> frame = nullptr;
+  std::map<int64_t, std::pair<std::unique_ptr<onnxruntime::ExecutionFrame>, size_t>>::iterator it;
+  size_t program_counter = 0;
+  //REVIEW(codemzs): We can get rid of this const cast by basically making partial graph exec manager APIs as SessionState.
+  PartialGraphExecutionManager& partial_graph_runs_manager = const_cast<SessionState&>(session_state).GetPartialGraphExecutionManager();
+  if (partial_graph_runs_manager.IsToBeExecutedRunId(run_id)) {
+    frame = std::shared_ptr<ExecutionFrame>(new ExecutionFrame(feed_mlvalue_idxs, feeds, fetch_mlvalue_idxs, fetches, fetch_allocators, session_state));
+    partial_graph_runs_manager.PreparePartialRun(frame, run_id, program_counter);
+  } else {
+    auto run = partial_graph_runs_manager.GetPartialRun(run_id);
+    (run.first)->UpdateFeeds(feed_mlvalue_idxs, feeds);
+    (run.first)->UpdateFetches(fetch_mlvalue_idxs, fetches);
+    program_counter = run.second;
+    frame = run.first;
+  }
+
+  // If the user has NOT requested to transfer ownership then reelease tensors that go out of scope starting from the last seen BreakOp.
+  // These tensors are BreakOp inputs that were given to the user as fetches in the last partial graph execution and it is possible
+  // they are are part of deallocation plan for the last seen BreakOp.
+  if ((program_counter > 0) &&
+      (session_state.GetKernel(exec_plan_vec[program_counter - 1].node_index)->KernelDef().OpName() == "BreakOp") &&
+      !session_state.GetTransferIntermediateTensorOwnership()) {
+    VLOGS(logger, 1) << "Releasing node ML values.";
+    ORT_RETURN_IF_ERROR(ReleaseNodeMLValues(*frame, seq_exec_plan, exec_plan_vec[program_counter - 1], logger));
+  }
+
+  // Determine partial graph execution start and end nodes.
+  size_t program_counter_end;
+  for (program_counter_end = program_counter;
+       (program_counter_end < exec_plan_last_index) &&
+       (session_state.GetKernel(exec_plan_vec[program_counter_end].node_index)->KernelDef().OpName() != "BreakOp");
+       program_counter_end += 1)
+    ;
+
+  ORT_ENFORCE(program_counter_end <= exec_plan_last_index);
+
+  bool last_node_break_op = session_state.GetKernel(exec_plan_vec[program_counter_end].node_index)->KernelDef().OpName() == "BreakOp";
+
+  // Do not execute if the first op of the graph is BreakOp (unlikely but legal corner case)
```

Comment:
Test coverage for this case? Do we also need to consider the case where it's the last node of the graph, or there are two consecutive yields?

Reply:
There is a test coverage for BreakOp being the first node but I'll add for BreakOp being the last one and two consecutives ones too. @mrry I plan to run a code coverage analyzer before check-in and will post the results in the PR summary section.